### PR TITLE
1077 Logout faster when session expires and when token doesnt work in api calls

### PR DESCRIPTION
**Pull request by DutchBen**

### CODE REVIEW

**Review by wouter1975 — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import { useGetOrchestratorConfig } from '@/hooks';
+
+export const WfoRouteChangeListener = () => {
+    const router = useRouter();
+    const { authActive } = useGetOrchestratorConfig();
+
+    useEffect(() => {
+        if (authActive) {
+            getSession().then((session) => {
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx:14]: Try to stick with the async await pattern, we have that in most places in this lib

File: packages/orchestrator-ui-components/src/rtk/api.ts [hunk 2]
```diff
@@ -85,6 +85,24 @@ export const orchestratorApi = createApi({
                 const graphqlFn = graphqlRequestBaseQuery({
                     url: customApi ? customApi.apiBaseUrl : graphqlEndpointCore,
                     prepareHeaders,
+                    customErrors: (error) => {
+                        const { name, message, stack, response } = error;
+                        if (
+                            response &&
+                            response.errors &&
+                            response.errors.length > 0
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/rtk/api.ts:93]: Can be replaced with `response?.errors?.length > 0`

**Review by ricardovdheijden — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                if (!session) {
+                    signOut();
+                }
+            });
+        }
+    }, [authActive, router]);
+    return <></>;
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx:21]: Looking at the usage of this component, you might want to add `children` here

File: packages/orchestrator-ui-components/src/rtk/api.ts [hunk 4]
```diff
@@ -85,6 +85,24 @@ export const orchestratorApi = createApi({
...
+                        ) {
+                            response.errors.map((error) => {
+                                if (
+                                    error.extensions?.error_type ===
+                                    'not_authorized'
```
Comment (Mark90) [inline packages/orchestrator-ui-components/src/rtk/api.ts:98]: I would suggest to change this to `not_authenticated` once it is available https://github.com/workfloworchestrator/orchestrator-core/issues/649

File: packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx (hunk 1 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WfoRouteChangeListener/WfoRouteChangeListener.tsx:14]: We have .thens in other places

File: packages/orchestrator-ui-components/src/rtk/api.ts (hunk 2 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/rtk/api.ts:93]: Fixd

File: packages/orchestrator-ui-components/src/rtk/api.ts (hunk 4 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/rtk/api.ts:98]: I created frontend ticket and added it as a todo https://github.com/workfloworchestrator/orchestrator-ui-library/issues/1105

**Pull request merged**